### PR TITLE
Add `dirMode` argument

### PR DESCRIPTION
**Pull request by jbsulli**
Wanted the ability to chmod on folders. 

This is just a quick flag I added. I don't know if it would be preferable to support a number/object to apply a different mode for folders but that could be added later without breaking functionality.

### CODE REVIEW

**Comment by sindresorhus** [comment]
I think we should follow [`vinyl-fs`](https://github.com/gulpjs/vinyl-fs/#optionsdirmode) here and have a separate argument just for directories, and make the first argument only apply to files.

**Comment by jbsulli** [comment]
OK, sounds good. I'll work on adding that functionality. Would you like passing `true` to still use mode settings for dirMode? Since mode only takes numbers and objects it would be easy to check for a boolean value.

**Comment by jbsulli** [comment]
It now supports separate dirMode value (either number or object value). Pass `true` to use mode argument on directories as well. Pass null or undefined mode to pass files through unchanged.

Note: `chmod(null, null)` would result in no permission changes.

File: index.js [hunk 1]
```diff
@@ -26,26 +26,48 @@ function normalize(mode) {
...
+		throw new TypeError('Expected dirMode to be null/undefined, true, a number, or an object');
 	}
 
 	return through.obj(function (file, enc, cb) {
-		if (file.isNull()) {
+		// default to file mode
+		var curMode = mode;
+		
+		// file is a directory?
+		if (file.isNull() && file.stat && file.stat.isDirectory && file.stat.isDirectory()) {
```
Comment (sindresorhus) [inline index.js]: The `file.stat.isDirectory` check is not needed. If `stat` exists, it exists.

File: index.js [hunk 2]
```diff
@@ -26,26 +26,48 @@ function normalize(mode) {
...
+	if (dirMode !== undefined && typeof dirMode !== 'number' && typeof dirMode !== 'object'){
+		throw new TypeError('Expected dirMode to be null/undefined, true, a number, or an object');
 	}
 
 	return through.obj(function (file, enc, cb) {
-		if (file.isNull()) {
+		// default to file mode
+		var curMode = mode;
+		
+		// file is a directory?
```
Comment (sindresorhus) [inline index.js]: Unnecessary comment.

Comment (sindresorhus) [inline index.js]: Actually, remove all the comments, they just say what the code already says.

File: readme.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ gulp.task('default', () => {
 
 ## API
 
-### chmod(mode)
+### chmod(mode, dirMode)
```
Comment (sindresorhus) [inline readme.md]: Should be marked as optional:

```
### chmod(mode, [dirMode])
```

**Comment by jbsulli** [comment]
Ok, those are taken care of. Comments I added just to help me think through the code and the additional check for stat was just because of what I saw in [gulp-zip](https://github.com/sindresorhus/gulp-zip/blob/master/index.js#L33).

I had one thing I wanted to check with you on. Before, you did a simple pass-through for any file that isNull(). Now, if it isNull() but not isDirectory() it would apply mode permissions. Do you see this as an issue? What is the desired behavior in that case?

**Comment by sindresorhus** [comment]
Looks good :)

**Comment by jbsulli** [comment]
Cool, thank you so much!

**Pull request merged**